.pre-commit-config.yaml

```diff
@@ -0,0 +1,12 @@
+# Automatically generated by nengo-bones, do not edit this file directly
+
+repos:
+-   repo: https://github.com/psf/black
+    rev: stable
+    hooks:
+    - id: black
+      exclude: |
+          (?x)(
+                docs/conf.py
+              | setup.py
+          )
```

Comment:
Would it be possible to do this through a `nengo-bones` command, rather than in every repo?  Something like `bones-pre-commit`, which would add our `nengo-bones` defined pre-commit hooks.  Advantages being that it only needs to be done once and then will work in all projects in that environment, reduces the config file clutter in downstream repos, and removes the need for the redundant `exclude` lists in `pre_commit_config_yaml` and `pyproject_toml`. But I'm not sure if it's possible to do it like that 🤷‍♂.

Reply:
Technically (at the moment) it isn't possible to not duplicate the `exclude` lists (see https://github.com/psf/black/issues/438). We could reduce the duplication by having a more general list that both files read, or be even more general with some kind of `black` config section that doesn't correspond to a file -- we could do similar things with `flake8` and whatnot as well, but those would be bigger changes to how Bones works so I think would require more thought.

I can't really see a way to implement something like `bones-pre-commit` that doesn't also involve having these files and running `pre-commit install` because `pre-commit install` makes changes to the current repository's metadata (specifically the `.git/hooks/` directory). If we wanted to manage the pre-commit stuff ourselves, we'd still have to have some kind of configuration and run `bones-pre-commit` in every repo. `pre-commit` is also a pretty big / mature / stable project in its own right (https://github.com/pre-commit/pre-commit) so reimplementing it doesn't feel like a good use of time.